src/test/java/org/sagebionetworks/bridge/udd/synapse/SynapseDownloadFromTableTaskCleanupFilesTest.java

```diff
@@ -0,0 +1,109 @@
+import org.sagebionetworks.bridge.udd.helper.MockFileHelper;
+
+// Deep tests for SynapseDownloadFromTableTask.cleanupFiles()
+public class SynapseDownloadFromTableTaskCleanupFilesTest {
+    private static final byte[] EMPTY_FILE_CONTENT = new byte[0];
+    private static final UploadSchemaKey TEST_SCHEMA_KEY = new UploadSchemaKey.Builder().withStudyId("test-study")
+            .withSchemaId("test-schema").withRevision(42).build();
+    private static final UploadSchema TEST_SCHEMA = new UploadSchema.Builder().withKey(TEST_SCHEMA_KEY)
+            .addField("foo", "STRING").build();
+
+    private File tmpDir;
+    private MockFileHelper mockFileHelper;
+    private SynapseDownloadFromTableTask task;
+
+    @BeforeMethod
+    public void setup() {
+        // mock file helper and temp dir
+        mockFileHelper = new MockFileHelper();
+        tmpDir = mockFileHelper.createTempDir();
+
+        SynapseDownloadFromTableParameters params = new SynapseDownloadFromTableParameters.Builder()
+                .withSynapseTableId("test-table-id").withHealthCode("test-health-code")
+                .withStartDate(LocalDate.parse("2015-03-09")).withEndDate(LocalDate.parse("2015-09-16"))
+                .withTempDir(tmpDir).withSchema(TEST_SCHEMA).build();
+        task = new SynapseDownloadFromTableTask(params);
+        task.setFileHelper(mockFileHelper);
+    }
+
+    @Test
+    public void noFiles() throws Exception {
+        // Default context has all null files.
+        executeTest();
+    }
+
+    @Test
+    public void csvOnly() throws Exception {
+        task.getContext().setCsvFile(createEmptyFile("csv.csv"));
+        executeTest();
+    }
+
+    @Test
+    public void csvAndBulkDownload() throws Exception {
+        task.getContext().setCsvFile(createEmptyFile("csv.csv"));
+        task.getContext().setBulkDownloadFile(createEmptyFile("download.zip"));
+        executeTest();
+    }
+
+    @Test
+    public void csvAndEditedCsv() throws Exception {
+        task.getContext().setCsvFile(createEmptyFile("csv.csv"));
+        task.getContext().setEditedCsvFile(createEmptyFile("csv-edited.csv"));
+        executeTest();
+    }
+
+    @Test
+    public void all3Files() throws Exception {
+        task.getContext().setCsvFile(createEmptyFile("csv.csv"));
+        task.getContext().setBulkDownloadFile(createEmptyFile("download.zip"));
+        task.getContext().setEditedCsvFile(createEmptyFile("csv-edited.csv"));
+        executeTest();
+    }
+
+    // branch coverage
+    @Test
+    public void nonNullFilesButDontExist() throws Exception {
+        // Create the files, but don't write any content to them, so they won't exist.
+        task.getContext().setCsvFile(mockFileHelper.newFile(tmpDir, "csv.csv"));
+        task.getContext().setBulkDownloadFile(mockFileHelper.newFile(tmpDir, "download.zip"));
+        task.getContext().setEditedCsvFile(mockFileHelper.newFile(tmpDir, "csv-edited.csv"));
+        executeTest();
+    }
+
```

Comment:
These tests would fail if there was a bug in the code. Specifically, if we didn't account for a particular file being null, or if we didn't account for a specific failure path. These tests have actually caught a bug, and writing these tests helped me catch another bug, so these tests definitely have value.